awx/ui_next/src/api/models/Credentials.js

```diff
@@ -15,10 +15,34 @@ class Credentials extends Base {
     });
   }
 
-  readInputSources(id, params) {
-    return this.http.get(`${this.baseUrl}${id}/input_sources/`, {
-      params,
-    });
+  readInputSources(id) {
+    const maxRequests = 5;
+    let requestCounter = 0;
+    const fetchInputSources = async (pageNo = 1, inputSources = []) => {
+      try {
+        requestCounter++;
+        const { data } = await this.http.get(
+          `${this.baseUrl}${id}/input_sources/`,
+          {
+            params: {
+              page: pageNo,
+              page_size: 200,
+            },
+          }
+        );
+        if (data.next && requestCounter <= maxRequests) {
+          return fetchInputSources(
+            pageNo + 1,
+            inputSources.concat(data.results)
+          );
+        }
+        return Promise.resolve(inputSources.concat(data.results));
+      } catch (error) {
+        return Promise.reject(error);
```

Comment:
Need to surface this error in [CredentialEdit.jsx#L154](https://github.com/ansible/awx/blob/devel/awx/ui_next/src/screens/Credential/CredentialEdit/CredentialEdit.jsx#L154)

```jsx
return <ContentError error={error} />;
```

Reply:
https://github.com/ansible/awx/pull/7281/commits/abb7486eb06399a7ce6facd516925118d6e02a30